pip2nix/models/package.py

```diff
@@ -67,13 +127,109 @@ def to_nix(self):
 
         args.update(self.raw_args)
 
+        # Prepare meta arguments.
+        meta_args = dict()
+        if include_lic:
+            license_nix = self.get_license_nix()
+            if license_nix:
+                meta_args['license'] = license_nix
+
         # Render name first
         raw_args = 'name = {};'.format(args.pop('name'))
         for k, v in sorted(args.items()):
             raw_args += '\n{} = {};'.format(k, v)
 
+        # Render meta arguments.
+        if meta_args:
+            raw_meta_args = ''
+            for k, v in sorted(meta_args.items()):
+                raw_meta_args += '{} = {};\n'.format(k, v)
+            meta = meta_template.format(meta_args=indent(2, raw_meta_args))
+            raw_args += '\n{}'.format(meta)
+
         return template.format(args=indent(2, raw_args))
 
+    def get_license_nix(self):
+        licenses = self.get_licenses_from_pkginfo()
+
+        # Convert license strings to nix.
+        nix_licenses = set()
+        for lic in licenses:
+            nix_licenses.add(license_to_nix(lic))
+
+        template = '[ {licenses} ]'
+        return template.format(licenses=' '.join(nix_licenses))
+
+    def get_licenses_from_setup(self):
```

Comment:
Can this be done with egg_info writers (like it works with tests_require.txt already)? Then it wouldn't be necessary to rerun the setup.py here.